ocaml/xapi/xapi_pool.ml

```diff
@@ -3385,6 +3385,18 @@ let remove_repository ~__context ~self ~value =
     (Db.Pool.get_repositories ~__context ~self) ;
   Db.Pool.remove_repositories ~__context ~self ~value
 
+let sync_with_yum_repos_in_progress = "sync_with_yum_repos_in_progress"
+
+let update_sync_status ~__context ~pool ~value =
+  if
+    List.mem_assoc sync_with_yum_repos_in_progress
+      (Db.Pool.get_other_config ~__context ~self:pool)
+  then
+    Db.Pool.remove_from_other_config ~__context ~self:pool
```

Comment:
What will be returned from `Db.Pool.remove_from_other_config` if the key doesn't exist? An exception?
Checking it firstly like this won't guarantee the key is there when doing the removal.
Remove it directly and catch the exception, or ignore anything it returned.

Reply:
If the kv pair with same key can be added multiple times, will the removing get only one removed when there are multiple pairs already in it?